engine.py

```diff
@@ -515,52 +510,118 @@ def has_ui(self):
     ##########################################################################################
     # logging
 
-    def log_debug(self, msg):
+    def _emit_log_message_FUTURE(self, handler, record):
         """
-        Log debug to the Maya script editor
+        Called by the engine to log messages in Maya script editor.
+        All log messages from the toolkit logging namespace will be passed to this method.
+
+        .. note:: This method will be called by the new logging system introduced in core v0.18.
+                  It will be enabled in January 2017 by removing the method name suffix "_FUTURE".
 
-        :param msg: The message to log
+        :param handler: Log handler that this message was dispatched from.
+                        Its default format is "[%(levelname)s %(basename)s] %(message)s".
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Standard python logging record.
+        :type record: :class:`~python.logging.LogRecord`
         """
-        global g_last_message_time
 
-        # get the time stamps
-        prev_time = g_last_message_time
-        current_time = time.time()
+        # Give a standard format to the message:
+        #     Shotgun DEBUG <basename>: <message>
+        # or
+        #     Shotgun <basename>: <message>
+        msg = "Shotgun %s%s: %s" % ("DEBUG " if record.levelno < logging.INFO else "",
+                                    record.basename,
+                                    record.message)
 
-        # update global counter
-        g_last_message_time = current_time
+        # Select Maya display function to use according to the logging record level.
+        if record.levelno < logging.WARNING:
+            fct = OpenMaya.MGlobal.displayInfo
+        elif record.levelno < logging.ERROR:
+            fct = OpenMaya.MGlobal.displayWarning
+        else:
+            fct = OpenMaya.MGlobal.displayError
+
+        # Display the message in Maya script editor in a thread safe manner.
+        self.async_execute_in_main_thread(fct, msg)
 
+    def log_debug(self, msg):
+        """
+        Logs debug messages in Maya script editor.
+
+        .. note:: This method is called by the legacy logging system.
+                  The new logging method will take over in January 2017
+                  while preserving backward compatibility.
+
+        :param msg: Debug message to log.
+                    This message has legacy format "%(basename)s: %(message)s".
+        """
+
+        # Do not log debug messages when debug logging setting is off.
         if not self.get_setting("debug_logging", False):
             return
 
-        msg = "Shotgun Debug [%0.3fs]: %s" % ((current_time-prev_time), msg)
+        # Give a standard format to the message:
+        #     Shotgun DEBUG <basename>: <message>
+        msg = "Shotgun DEBUG %s" % msg
```

Comment:
OK. Debug message timing is back.

It was removed based on a comment from @josh-t in a past CR, since timing is already logged better in `tk-maya.log`.